src/QmlControls/QGroundControlQmlGlobal.h

```diff
@@ -145,6 +145,22 @@ class QGroundControlQmlGlobal : public QGCTool
 
     QString appSettingsDistanceUnitsString(void) const { return FactMetaData::appSettingsDistanceUnitsString(); }
 
+    /// Converts from meters to the user specified distance unit
+    Q_INVOKABLE QVariant metersToAppSettingsAltitudeUnits(const QVariant& meters) const { return FactMetaData::metersToAppSettingsAltitudeUnits(meters); }
```

Comment:
This is becoming ugly and turning `QGroundControlQmlGlobal.h` into a blob. This is needed though when QML calculates altitudes (as opposed to showing values stored as `Facts`) - I think it has every privilege to want to do so and it is correct to supply QML with functions like these. But maybe it's time to wrap them into a `UnitCoversion` class that `QGroundControlQmlGlobal` exposes.

Reply:
I don't disagree. Will need to pull the string on usage of it through all the Qml.

Reply:
Since you asked me to rename existing methods (e.g.: ["distance" to "horizontalDistance"](https://github.com/mavlink/qgroundcontrol/pull/8692/commits/1e98a7abc8da7f14fb348880ad43b4435739413a#diff-b4d15c63e4db01b93ae2cb70e47602d1R64)) I had to update Qml so i did take the liberty of wrapping all those unit methods in `QGroundControlQmlGlobal` into [QmlUnitsConversion](https://github.com/mavlink/qgroundcontrol/pull/8692/commits/1e98a7abc8da7f14fb348880ad43b4435739413a#diff-d0673baf68efbb0ab1f10f243c8dce5aR1) and diligently chased all qml references.